### PR TITLE
Upgrade to Embedded hal 1.0.0

### PR DESCRIPTION
**Pull request by PixmaHestiia**
Upgrade deps for embedded-hal 1.0.0.

### CODE REVIEW

**Review by Rahix — APPROVED** [review]
Very nice, thanks a lot.  I'm currently still holding off on merging this PR because it means all HALs still on 0.2.x will no longer be supported.  Trying to coordinate with other people right now to ensure a smooth transition and will merge once everyone is ready.

**Comment by Rahix** [comment]
Seems we are good to go :)))

**Comment by blaa** [comment]
> Very nice, thanks a lot. I'm currently still holding off on merging this PR because it means all HALs still on 0.2.x will no longer be supported. Trying to coordinate with other people right now to ensure a smooth transition and will merge once everyone is ready.

I think this is compat layer that is supposed to fix this particular problem:
https://github.com/ryankurte/embedded-hal-compat

**Pull request merged**
